src/jobspy/scrapers/linkedin/__init__.py

```diff
@@ -236,7 +236,7 @@ def _get_job_details(self, job_id: str) -> dict:
         """
         try:
             response = self.session.get(
-                f"{self.base_url}/jobs-guest/jobs/api/jobPosting/{job_id}", timeout=5
+                f"{self.base_url}/jobs/view/{job_id}", timeout=5
```

Comment:
@cullenwatson This doesn't work.

It breaks all the code below where soup wont have the classes and soup.find doesn't find those mentioned classname.

And without login into linkedin, you can't view job id with '/job/view/{job_id}', so you cant scrape anything , as it redirects to login.

Reverse this push request, as it will fail for the whole linkedin description for everyone

You should have tested if it works.

Reply:
i did and it works

Reply:
Ohh Sryy for that, Just tried and it works, but:

```python
        soup = BeautifulSoup(response.text, "html.parser")
        div_content = soup.find(
            "div", class_=lambda x: x and "show-more-less-html__markup" in x
        )
        description = None
        if div_content is not None:
            div_content = remove_attributes(div_content)
            description = div_content.prettify(formatter="html")
            if self.scraper_input.description_format == DescriptionFormat.MARKDOWN:
                description = markdown_converter(description)
        h3_tag = soup.find(
            "h3", text=lambda text: text and "Job function" in text.strip()
        )
        job_function = None
        if h3_tag:
            job_function_span = h3_tag.find_next(
                "span", class_="description__job-criteria-text"
            )
            if job_function_span:
                job_function = job_function_span.text.strip()
        return {
            "description": description,
            "job_level": self._parse_job_level(soup),
            "company_industry": self._parse_company_industry(soup),
            "job_type": self._parse_job_type(soup),
            "job_url_direct": self._parse_job_url_direct(soup),
            "logo_photo_url": soup.find("img", {"class": "artdeco-entity-image"}).get(
                "data-delayed-url"
            ),
            "job_function": job_function,
        }

```

In the above code, for the `soup.find()`
`show-more-less-html__markup`
`Job function`

These are not present in   `f"{self.base_url}/jobs/view/{job_id}", timeout=5`  html code. Don't know how you are getting them and how its working.